dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -390,71 +447,23 @@ public String toJson(final Contentlet contentlet) throws DotMappingException {
-
-		//Verify if it is enabled the option to regenerate missing metadata files on reindex
-		final boolean regenerateMissingMetadata = Config
-				.getBooleanProperty("regenerate.missing.metadata.on.reindex", true);
-
-		/*
-		Verify if it is enabled the option to always regenerate metadata files on reindex,
-		enabling this could affect greatly the performance of a reindex process.
-		 */
-		final boolean alwaysRegenerateMetadata = Config
-				.getBooleanProperty("always.regenerate.metadata.on.reindex", false);
-
-		Map<String, Object> metadataMap = null;
-
-		if (contentlet.isLive() || contentlet.isWorking()) {
-
-			final Optional<com.dotcms.contenttype.model.field.FieldVariable> customIndexMetaDataFieldsOpt =
-					FactoryLocator.getFieldFactory().byFieldVariableKey(field.getIdentifier(), BinaryField.INDEX_METADATA_FIELDS);
-			final TikaUtils tikaUtils = new TikaUtils();
-
-			final Set<String> metadataFields = customIndexMetaDataFieldsOpt.isPresent()?
-					new HashSet<>(Arrays.asList(customIndexMetaDataFieldsOpt.get().value().split(StringPool.COMMA))):
-					tikaUtils.getConfiguredMetadataFields();  // gets the metadata fields to filter from the tika metadata
-
-			final File binaryField = contentlet.getBinary(field.getVelocityVarName());
-			if (null != binaryField && binaryField.exists() && binaryField.canRead()) {
-
-				if (alwaysRegenerateMetadata) {
-					metadataMap = tikaUtils.generateMetaDataForce(contentlet, binaryField, field.getVelocityVarName(), metadataFields);
-				} else if (regenerateMissingMetadata) {
-					metadataMap = tikaUtils.generateMetaData(contentlet, binaryField, field.getVelocityVarName(), metadataFields);
-				}
-			}
-
-			// see if we have content metadata
-			if(null != metadataMap) {
-
-				for (final String metadataKey : metadataFields) {
-
-					final Object metadataValue = metadataMap.get(metadataKey);
-					if (null != metadataValue) {
-						mapLowered.put(FileAssetAPI.META_DATA_FIELD.toLowerCase() + StringPool.PERIOD + metadataKey.toLowerCase(), metadataValue);
-						if (metadataKey.contains(FileAssetAPI.CONTENT_FIELD)) {
-							stringWriter.append(metadataValue.toString()).append(' ');
-						}
-					}
-				}
-			}
-		}
-	}
-
-	private Optional<Field> findFirstBinaryFieldIndexable(final Contentlet contentlet) {
-
-		final List<Field> fields  = new ArrayList<>(
-				FieldsCache.getFieldsByStructureInode(contentlet.getStructureInode()));
-
-		return fields.stream().filter(Field::isIndexed)
-				.filter(field -> field.getFieldType().equals(Field.FieldType.BINARY.toString()))
-				.findFirst();
-	}
 
+    private Object preProcessValue(final String compositeKey, final Object value) {
+        if ("metadata.content".equals(compositeKey)) {
+            if (null == value) {
+                //This "NO_METADATA" constant is getting relocated from tika utils
+                return NO_METADATA;
+            }
+            //TODO:make sure this is not a dupe META_DATA_MAX_SIZE
```

Comment:
when is the TODO going to be implemented?

Reply:
done!. Comment updated